features/post/post-card/post-info-section/post-info-section.tsx

```diff
@@ -0,0 +1,114 @@
+import React, { ReactNode } from 'react';
+
+import { BookmarkOutline } from '@/assets/icons/bookmark-outline';
+import { HeartFilled } from '@/assets/icons/heart-filled';
+import { HeartOutlined } from '@/assets/icons/heart-outlined';
+import { PaperPlaneOutline } from '@/assets/icons/paper-plane-outline';
+import { DateFormatter, useDateFormat } from '@/shared/hooks/useDateFormat';
+import { useTranslation } from '@/shared/hooks/useTranslation';
+import { Avatar, IconButton, Typography } from '@/shared/ui';
+import { useRouter } from 'next/router';
+
+import s from './post-info.module.scss';
+
+type PostInfoProps = {
+  actions?: ReactNode;
+  createdAt: string;
+  icons?: {
+    avatar: string;
+    userName: string;
+  }[];
+  likesCount: number;
+};
+
+export const PostInfoSection = (props: PostInfoProps) => {
+  const { actions, createdAt, icons, likesCount } = props;
+  const { locale } = useRouter();
+  const { t } = useTranslation();
+
+  const customDateTimeFormatter: DateFormatter = parts => {
+    const date = new Date(createdAt);
+
+    return new Intl.DateTimeFormat(locale, {
+      day: 'numeric',
+      month: 'long',
+      year: 'numeric',
+    }).format(date);
+  };
+
+  const formattedDate = useDateFormat(
```

Comment:
This partially duplicates the internal logic of the `useDateFormat` hook unnecessarily. However, it also uses the `format` method of the `Intl.DateTimeFormat` instance, which is not in the hook, indicating that this hook is simply not being used as intended here.
The best solution is to rewrite the hook.
1. The hook should under the hood call `useRouter`, create an instance of `Intl.DateTimeFormat` and simply return that object. Thus, the hook should accept the same parameters as the `Intl.DateTimeFormat` constructor. As such, it is suitable for use in this component.
2. Rename the old `useDateFormat` to `useDateFormatParts` and rewrite it to use the new `useDateFormat` hook. Leave the `useDateFormatParts` api unchanged for compatibility!